### PR TITLE
First pass at chat refactor

### PR DESCRIPTION
**Pull request by jordanreedie**
## Description
This set of changes separates out much of the functionality of the chat page into logically separate pieces & components. At a high level we now have separate Lobby and Meeting components, removing the requirement to check if a user is in a room or not before rendering anything in every component.

For the most part, everything is still connected to redux at the top level, which means we need a bunch of props just for plumbing. This is annoying, and next pass will involve redoing the redux connection(s) and also hopefully removing as many unnecessary pieces of state as possible.

Also, we have a LOT of side effects in our actions. We may want to consider using something that is set up for handling them, like redux-thunk or redux-saga.

I've left a bunch of FIXME / TODO comments in there which I will need to address in the next pass - feel free to comment on them, or not.

Given the size / scope of the changes, it might be easiest to review them over a riff chat or something

## Motivation and context
This addresses _Chat: Refactoring for style and modularity, First Half_ #72

## How has this been tested?
Tested locally, however changes will need to be tested more thoroughly with a few willing guinea pigs :)
It is very possible there are a few remaining bugs that I haven't caught

## Types of changes

- ✅ Refactor (non-breaking change which adds functionality)

## Checklist:

- ✅ My code follows the code style of this project. **[REQUIRED]** (mostly)
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

File: src/redux/reducers/chat.js [hunk 1]
```diff
@@ -35,226 +35,257 @@ import {
...
-    config: null,
-    signalMasterPath: ''
-  },
-  textchat: {
-    messages: [],
-    lastRoom: "",
-    lastMeetingId: "",
-    badge: 0
-  }
+    peerColors: ['#f56b6b', '#128EAD', '#7caf5f', '#f2a466', '#321325', '#3C493F', '#1B998B'],
```
Comment (jordanreedie) [inline src/redux/reducers/chat.js]: i'm realizing now i didn't mean to convert this file from 2 -> 4 spaces, whoops

File: src/components/Chat/Meeting/MeetingRoom/RemoteVideoContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            );
+        } else {
+            return this.collectPeerVideos();
+        }
+    }
+
+    render() {
+        return (
+            <div className="remotes" id="remoteVideos">
+                <div ref="remotes" className="columns is-multiline is-centered is-mobile">
```
Comment (mlippert) [inline src/components/Chat/Meeting/MeetingRoom/RemoteVideoContainer.jsx]: I can't see that this `ref` is used anywhere, and it seems to be a legacy ref which we shouldn't be using anyway (see https://reactjs.org/docs/refs-and-the-dom.html)

File: src/components/Chat/styled.js [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+const MainContent = styled.div`
 padding: 1.5rem;
 `;
 
-export const VideoPlaceholder = styled.div.attrs({
-  className: 'has-text-centered',
-  ref: 'local'
+const Placeholder = styled.div.attrs({
+    className: 'has-text-centered',
+    ref: 'local'
```
Comment (mlippert) [inline src/components/Chat/styled.js]: I can't see that this `ref` is used anywhere, and it seems to be a legacy ref which we shouldn't be using anyway (see https://reactjs.org/docs/refs-and-the-dom.html)

File: src/components/Chat/Meeting/MeetingRoom/MeetingRoom.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        // in which case we probably should still display the empty room, no?
+        if (this.props.webRtcPeers.length > 0 || this.props.webRtcRemoteSharedScreen) {
+            return (
+                <div
+                    className='column'
+                    tabIndex='0'
+                    aria-label={`Video chat. ${readablePeers(this.props.webRtcPeers)}`}
+                >
+                    <RemoteVideoContainer
+                        ref='remote'
```
Comment (mlippert) [inline src/components/Chat/Meeting/MeetingRoom/MeetingRoom.jsx]: I can't see that this `ref` is used anywhere, and it seems to be a legacy ref which we shouldn't be using anyway (see https://reactjs.org/docs/refs-and-the-dom.html)

File: src/redux/reducers/chat.js (hunk 1 above)
Comment (mlippert) [reply src/redux/reducers/chat.js]: I fixed this for you

**Comment by mlippert** [comment]
**Release** [1.2.0-dev.7](https://github.com/rifflearning/riff-rtc/releases/tag/1.2.0-dev.7)

**Pull request merged**